lib/chef/resource/launchd.rb

```diff
@@ -0,0 +1,285 @@
+        set_or_return(:nice, nice, :kind_of => [TrueClass, FalseClass])
+      end
+
+      def on_demand(on_demand = nil)
+        set_or_return(:on_demand, on_demand, :kind_of => String)
+      end
+
+      def username(username = nil)
+        set_or_return(:username, username, :kind_of => String)
+      end
+
+      def group(group = nil)
+        set_or_return(:group, group, :kind_of => String)
+      end
+
+      def inetd_compatibility(inetd_compatibility = nil)
+        set_or_return(
+          :inetd_compatibility, inetd_compatibility, :kind_of => Hash
+        )
+      end
+
+      def init_groups(init_groups = nil)
+        set_or_return(:init_groups, init_groups, :kind_of => [TrueClass, FalseClass])
+      end
+
+      def process_type(process_type = nil)
+        set_or_return(:process_type, process_type, :kind_of => String)
+      end
+
+      def program(program = nil)
+        set_or_return(:program, program, :kind_of => String)
+      end
+
+      def program_arguments(program_arguments = nil)
+        set_or_return(:program_arguments, program_arguments, :kind_of => Array)
+      end
+
+      def queue_directories(queue_directories = nil)
+        set_or_return(:queue_directories, queue_directories, :kind_of => Array)
+      end
+
+      def root_directory(root_directory = nil)
+        set_or_return(:root_directory, root_directory, :kind_of => String)
+      end
+
+      def run_at_load(run_at_load = nil)
+        set_or_return(
+          :run_at_load, run_at_load, :kind_of => [TrueClass, FalseClass]
+        )
+      end
+
+      def sockets(sockets = nil)
+        set_or_return(:sockets, sockets, :kind_of => Hash)
+      end
+
+      def soft_resource_limits(soft_resource_limits = nil)
+        set_or_return(
+          :soft_resource_limits, soft_resource_limits, :kind_of => Array
+        )
+      end
+
+      def standard_error_path(standard_error_path = nil)
+        set_or_return(
+          :standard_error_path, standard_error_path, :kind_of => String
+        )
+      end
+
+      def standard_in_path(standard_in_path = nil)
+        set_or_return(:standard_in_path, standard_in_path, :kind_of => String)
+      end
+
+      def standard_out_path(standard_out_path = nil)
+        set_or_return(:standard_out_path, standard_out_path, :kind_of => String)
+      end
+
+      def start_calendar_interval(start_calendar_interval = nil)
+        set_or_return(
+          :start_calendar_interval, start_calendar_interval, :kind_of => Hash
+        )
+      end
+
+      def start_interval(start_interval = nil)
+        set_or_return(:start_interval, start_interval, :kind_of => Integer)
+      end
+
+      def start_on_mount(start_on_mount = nil)
+        set_or_return(
+          :start_on_mount, start_on_mount, :kind_of => [TrueClass, FalseClass]
+        )
+      end
+
+      def throttle_interval(throttle_interval = nil)
+        set_or_return(
+          :throttle_interval, throttle_interval, :kind_of => Integer
+        )
```

Comment:
it's weird to me visually that a bunch of these aren't on one line. I get that with whitespace they're over 80 characters though. Not saying to change it, I just kept expecting these ones to be different somehow from the ones on a single line.